packages/web3-core/types/index.d.ts

```diff
@@ -442,8 +442,8 @@ export interface RequestArguments {
 }
 
 export interface AbstractProvider {
-    sendAsync(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
-    send?(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
+    sendAsync(payload: JsonRpcPayload, callback?: (error: Error | null, result?: JsonRpcResponse) => Promise<any> | void): void;
```

Comment:
Could this be typed as `Promise<unknown>`?

Reply:
It's updated ✅ 
And for request method, why do we let any as promise return type ? Isn't it better to put unknown too ? since conventionally it is advisable to put an unknown to a return value which will be assigned a new type in output, so we are obliged to assign a type to it in order to be able to use it contrary to any which allows any manipulation, it encourages good practice, doesn't it?

What do you think about it? 😄

Reply:
I agree!  `unknown` is generally better than `any` (except for special cases, but i don't believe this to be one of them).

Reply:
For my personal knowledge, can u explain me why ? 😄

Reply:
I would defer to others who have explained it thoroughly and more eloquently than me -- for example:
- https://mainawycliffe.dev/blog/typescript-use-unknown-instead-of-any/
- https://www.benmvp.com/blog/when-use-typescript-unknown-versus-any/
- https://dev.to/tomdohnal/typescript-any-vs-unknown-a-deep-dive-3iem 

:)

Reply:
Thanks for sharing, I will take a look at it ;)